src/OrchardCore.Modules/OrchardCore.AuditTrail/Drivers/AuditTrailPartDisplayDriver.cs

```diff
@@ -0,0 +1,37 @@
+﻿using OrchardCore.AuditTrail.Models;
+using OrchardCore.AuditTrail.Settings;
+using OrchardCore.AuditTrail.ViewModels;
+using OrchardCore.ContentManagement.Display.ContentDisplay;
+using OrchardCore.ContentManagement.Display.Models;
+using OrchardCore.DisplayManagement.ModelBinding;
+using OrchardCore.DisplayManagement.Views;
+using System.Threading.Tasks;
+
+namespace OrchardCore.AuditTrail.Drivers
+{
+    public class AuditTrailPartDisplayDriver : ContentPartDisplayDriver<AuditTrailPart>
+    {
+        public override IDisplayResult Edit(AuditTrailPart part, BuildPartEditorContext context)
+        {
+            var settings = context.TypePartDefinition.GetSettings<AuditTrailPartSettings>();
+            if (settings.ShowAuditTrailCommentInput)
+            {
+                return Initialize<AuditTrailCommentViewModel>($"{nameof(AuditTrailPart)}_Comment", model =>
```

Comment:
This should use `GetEditorShapeType(context)` and have a default `.Option` for the editor modes.

Reply:
What do you mean, by `.Option`, exactly? @domonkosgabor did you do anything on this front?

Reply:
Ah I see.